### DISCUSSION TITLE
jump-to-definition not working for dependencies that don't have a *.sources.jar

### DISCUSSION THREAD

**Discussion by scottyj503** [scalameta/metals#3850] [General]
When I attempt to "jump-to" a dependency that does not have a *.sources.jar file (.metals/readonly.dependencies/), the feature does not work.  For those dependencies that do have a *.sources.jar, the feature works as expected.  

The public repo for those said dependencies do not have a *.sources.jar file available.

With the info I was able to approximate from my research, it _appears_ that Metals does not currently have a way to de-compile from non-source jars.  Is that correct?

Or, have I missed a setting/command that would solve this problem?

I am using metals-emacs: 0.11.2
sbt: 1.6.1

~ Thanks!!

**Comment by kpodsiad** [comment]
It's true, jump-to-definition doesn't work when the target has no sources. There's a feature request about [decompiling .class files](https://github.com/scalameta/metals-feature-requests/issues/210) which will unblock this scenario.
